crates/re_ui/src/lib.rs

```diff
@@ -515,6 +515,64 @@ impl ReUi {
         response
     }
 
+    /// Popup similar to [`egui::popup_below_widget`] but suitable for use with
+    /// [`crate::list_item::ListItem`].
+    pub fn list_item_popup<R>(
+        ui: &egui::Ui,
+        popup_id: egui::Id,
+        widget_response: &egui::Response,
+        vertical_offset: f32,
+        add_contents: impl FnOnce(&mut egui::Ui) -> R,
+    ) -> Option<R> {
+        if !ui.memory(|mem| mem.is_popup_open(popup_id)) {
+            return None;
+        }
+
+        let pos = widget_response.rect.left_bottom() + egui::vec2(0.0, vertical_offset);
+        let pivot = Align2::LEFT_TOP;
+
+        let inner = egui::Area::new(popup_id)
+            .order(egui::Order::Foreground)
+            .constrain(true)
+            .fixed_pos(pos)
+            .pivot(pivot)
+            .show(ui.ctx(), |ui| {
+                let frame = egui::Frame {
+                    fill: ui.visuals().panel_fill,
+                    ..Default::default()
+                };
+                let frame_margin = frame.total_margin();
+                frame
+                    .show(ui, |ui| {
+                        ui.with_layout(egui::Layout::top_down_justified(egui::Align::LEFT), |ui| {
+                            ui.set_width(widget_response.rect.width() - frame_margin.sum().x);
+
+                            ui.set_clip_rect(ui.cursor());
+
+                            egui::ScrollArea::vertical()
+                                .show(ui, |ui| {
+                                    egui::Frame {
+                                        //TODO(ab): use design token
+                                        inner_margin: egui::Margin::symmetric(8.0, 0.0),
+                                        ..Default::default()
+                                    }
+                                    .show(ui, add_contents)
+                                    .inner
+                                })
+                                .inner
+                        })
+                        .inner
+                    })
+                    .inner
+            })
+            .inner;
+
+        if ui.input(|i| i.key_pressed(egui::Key::Escape)) || widget_response.clicked_elsewhere() {
+            ui.memory_mut(|mem| mem.close_popup());
+        }
+        Some(inner)
```

Comment:
just to explain my alternative to `.inner`x5:

```suggestion
        let pos = widget_response.rect.left_bottom() + egui::vec2(0.0, vertical_offset);
        let pivot = Align2::LEFT_TOP;

        let mut ret = None;

        let inner = egui::Area::new(popup_id)
            .order(egui::Order::Foreground)
            .constrain(true)
            .fixed_pos(pos)
            .pivot(pivot)
            .show(ui.ctx(), |ui| {
                let frame = egui::Frame {
                    fill: ui.visuals().panel_fill,
                    ..Default::default()
                };
                let frame_margin = frame.total_margin();
                frame
                    .show(ui, |ui| {
                        ui.with_layout(egui::Layout::top_down_justified(egui::Align::LEFT), |ui| {
                            ui.set_width(widget_response.rect.width() - frame_margin.sum().x);

                            ui.set_clip_rect(ui.cursor());

                            egui::ScrollArea::vertical()
                                .show(ui, |ui| {
                                    egui::Frame {
                                        //TODO(ab): use design token
                                        inner_margin: egui::Margin::symmetric(8.0, 0.0),
                                        ..Default::default()
                                    }
                                    .show(ui, |ui| {
                                        ret = Some(add_contents(ui));
                                    });
                                });
                        });
                    });
            });

        if ui.input(|i| i.key_pressed(egui::Key::Escape)) || widget_response.clicked_elsewhere() {
            ui.memory_mut(|mem| mem.close_popup());
        }
        
        ret
```

wether or not it is preferable is a style choice 🤷

Reply:
That somehow reduced the rightward shift, so that's nice 👍🏻